### DISCUSSION TITLE
Eloquent Driver Pros & Cons

### DISCUSSION THREAD

**Discussion by webzes** [statamic/cms#9904] [General]
Hello all

I am looking to start my next project and I am hoping to use Statamic as the CMS.

I would like to use the eloquent driver for collection content as this would allow me to utilize some native Laravel packages that require adding traits to the models etc...

I have seen somewhere that the eloquent driver for Statamic does not support eager loading relationships, is this true?

Also is there a document somewhere that lists the pros and cons of using either flat file vs eloquent? I am fully aware flat file would be faster, I am just more interested in a comparison of functionality.

I have seen the Runway plugin but for now, I am looking to stay first-party as much as possible.

Thanks!

**Comment by joshuablum** [comment]
Hey @webzes,

> I would like to use the eloquent driver for collection content as this would allow me to utilize some native Laravel packages that require adding traits to the models etc...

Feel free to use the Eloquent Driver _just_ for collection content. You can configure what's stored in files and what's in the database [in the driver's config](https://github.com/statamic/eloquent-driver/blob/master/config/eloquent-driver.php). On top of that, the Eloquent Driver supports importing from files _and_ exporting back to flat files at any time. So you can just give it a try, see [here](https://github.com/statamic/eloquent-driver?tab=readme-ov-file#importing-existing-file-based-content).

> I have seen somewhere that the eloquent driver for Statamic does not support eager loading relationships, is this true?

Yes, iirc.

> Also is there a document somewhere that lists the pros and cons of using either flat file vs eloquent? I am fully aware flat file would be faster, I am just more interested in a comparison of functionality.

No, there is no such document (yet). We usually recommend to **stay flat file as long as it works for you** or specific requirements of your project require a database. Again, you can move to the database (or back) at any point.

> I have seen the Runway plugin but for now, I am looking to stay first-party as much as possible.

Runway is an awesome addon and might be even better for your project depending on your needs. And it's part of the [Rad Pack](https://statamic.com/blog/rad-pack) and the creator, @duncanmcclean, joined Statamic and is part of the Core Team. So no need to worry about it not being "first-party".

**Reply by ryanmitchell** [reply]
FWIW I help maintain the eloquent driver, and I always reach for Runway before I consider going to eloquent. More often than not its one or two collections that require database support and Runway fits the bill perfectly for that.

**Comment by webzes** [comment]
Great, thanks for your replies. I did not realise Rad Pack was part of the core team so that gives me the confidence to use Runway for such an important part of the CMS 👍

**Comment by vitalijalbu** [comment]
Hi guys, another question: if I plan to have 10k entries, in this case i think mysql is the best solution.
But I don’t understand few things:
- if i move to mysql driver, statamic continuously will generate md files and then they will be synced to the db? Or it will use db only for manipulating data?
- if statamic tomorrow adds or changes a field name, should i update also the db schema? I will loose my data  in the db?

**Reply by ryanmitchell** [reply]
No it uses the database as the store for entries or whatever other repositories you enable. 

The blueprint fields are stored in a json column so you can modify the blueprint and the fields will ‘just work’.

**Reply by vitalijalbu** [reply]
Ok, thanks. But why they just don’t do it natively? Without any library?

**Reply by ryanmitchell** [reply]
I don't understand your question. The eloquent driver is a 1st party add-on, its officially supported by Statamic.
